### ISSUE TITLE
When searching for part of the string it returns results only in specific case

### ISSUE THREAD

**Issue by Denis2310** [Smile-SA/elasticsuite#3328]
I have product with name "Rotational Viscometer". When I search for "rota" nothing is shown in autocomplete, after that when I search for "rotat" it returns results, then again "rotati" doesnt return anything.

How is it possible that there are no results for "rota" then there are results for "rotat" and then again nothing for "rotati".

### Preconditions

Magento Version : 2.4.6-p4

ElasticSuite Version : 2.11.6

Environment : Developer mode

Third party modules :

### Steps to reproduce

1. Go to magento search
2. Type part of the word, in my case "rota", "rotat" and "rotati" instead of "rotational"
3. There are results only for substrings "rotat" and "rotation"

### Expected result
1. Search for "rot" - results are found 
2. Search for "rota"  results are found
3. Search for "rotat" results are found
4. Search for "rotati" results are found

### Actual result
1. Search for "rot" - results are not found 
2. Search for "rota"  results are not found
3. Search for "rotat" **results are found**
4. Search for "rotati" results are not found

- Basically I get results for: "rotat", "rotation", and full word "rotational"
- In autocomplete there are no results for "rot", "rota", there are for "rotat" and then back again nothing for "rotati" etc.
- When I click enter then there is a message "No exact results found for: .." and some products are shown, and for "rotat" there is no that message and correct products are shown.

**Comment by vahonc** [comment]
Hello @Denis2310,

First of all, when you create a new issue, you should use the search, because perhaps such questions have already been raised here.

Look at these issue [#2600](https://github.com/Smile-SA/elasticsuite/issues/2600#issuecomment-1157667466), [#2811](https://github.com/Smile-SA/elasticsuite/issues/2811), [#3070](https://github.com/Smile-SA/elasticsuite/issues/3070) [#3104](https://github.com/Smile-SA/elasticsuite/issues/3104).

BR,
Vadym

**Comment by Denis2310** [comment]
@vahonc 
I understand what you mean but I dont see that any of these issues has been fixed.. There are some suggestions to change some configuration in backend but it doesnt fix the issue.

**Comment by Denis2310** [comment]
Resolved here: https://github.com/Smile-SA/elasticsuite/issues/3329
